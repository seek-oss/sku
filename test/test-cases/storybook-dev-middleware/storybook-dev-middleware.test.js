const path = require('path');
const { promisify } = require('util');
const rimrafAsync = promisify(require('rimraf'));
const fs = require('fs-extra');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const waitForUrls = require('../../utils/waitForUrls');
const { getPathFromCwd } = require('../../../lib/cwd');
const appDir = path.resolve(__dirname, 'app');

async function createPackageLink(name) {
  await fs.mkdirp(`${__dirname}/app/node_modules`);
  await fs.symlink(
    getPathFromCwd(`node_modules/${name}`),
    `${__dirname}/app/node_modules/${name}`,
  );
}

async function setUpLocalDependencies() {
  const nodeModules = `${__dirname}/app/node_modules`;
  await rimrafAsync(nodeModules);
  await Promise.all(['react', 'react-dom'].map(createPackageLink));
}

describe('storybook-dev-middleware', () => {
  beforeAll(async () => {
    // "Install" React and braid-design-system into this test app so that webpack-node-externals
    // treats them correctly.
    await setUpLocalDependencies();
  });

  describe('storybook', () => {
    const storybookUrl = `http://localhost:8081`;
    let process;

    beforeAll(async () => {
      process = await runSkuScriptInDir('storybook', appDir);
      await waitForUrls(storybookUrl, `${storybookUrl}/test-middleware`);
    });

    afterAll(async () => {
      await process.kill();
    });

    it('should support the supplied middleware', async () => {
      const snapshot = await getAppSnapshot(`${storybookUrl}/test-middleware`);
      expect(snapshot).toMatchSnapshot();
    });
  });
});
