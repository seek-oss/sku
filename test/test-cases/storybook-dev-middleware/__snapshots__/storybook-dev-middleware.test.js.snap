// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storybook-dev-middleware storybook should support the supplied middleware 1`] = `
"OK"
POST HYDRATE DIFFS: 
-OK
\\ No newline at end of file
+<html>
+  <head>
+  </head>
+  <body>
+    OK
+  </body>
+</html>
\\ No newline at end of file

`;
